### ISSUE TITLE
ZodUnion options type bound causes unwanted errors

### ISSUE THREAD

**Issue by jedwards1211** [colinhacks/zod#3812]
This shouldn't happen:

```ts
z.union(
  [1, 2, 3].map((value) => z.literal(value)),
  // ERROR: Argument of type 'ZodLiteral<number>[]' is not assignable to parameter of type 'readonly [ZodTypeAny, ZodTypeAny, ...ZodTypeAny[]]'.
  //   Source provides no match for required element at position 0 in target.ts(2345)
)
```

I get that `[ZodTypeAny, ZodTypeAny, ...ZodTypeAny[]]` is supposed to catch cases where someone accidentally makes a zero or one-option union, but is it worth the hassle it causes in cases like this?  It's pretty annoying having to manually suppress this error.

The behavior of zero and one-element unions is well defined and there are even cases where it's perfectly reasonable to let code that creates a schema dynamically create a zero or one-element union.  Please make the type parameter just `ZodTypeAny[]`.

**Comment by kockar96** [comment]
You can write your own custom function to achieve this.
As example:

const SUPPORTED_NUMBERS = [1, 2, 3, 5, 8, 11] as const;

export function constructZodLiteralUnionType<T extends z.ZodLiteral<unknown>>(
  literals: T[],
) {
  if (!isValidZodLiteralUnion(literals)) {
    throw new Error(
      'Literals passed do not meet the criteria for constructing a union schema, the minimum length is 2',
    );
  }
  return z.union(literals);
}


export const NumberSchema = constructZodLiteralUnionType(
  SUPPORTED_NUMBERS.map((literal) => z.literal(literal)),
);

type Number = z.infer<typeof NumberSchema>;
         // ^ type Number = 2 | 1 | 3 | 5 | 8 | 11

**Comment by jedwards1211** [comment]
Of course, but the core type parameter doesn't need to be so complicated
